content/posts/jenkins-x-flow-integration.md

```diff
@@ -0,0 +1,147 @@
+     - sh: jx step post build --image $DOCKER_REGISTRY/$ORG/$APP_NAME:\$(cat VERSION)
+       name: post-build
+   promote:
+     steps:
+     - sh: jx step changelog --version v\$(cat ../../VERSION)
+       name: changelog
+     - comment: call CloudBees Flow to run a release
+       sh: >
+         curl -X POST --header "Authorization: Basic $(jx step credential -s flow-token -k token)" --header "Content-Type: application/json" --header "Accept: application/json" -d "{}" "https://ps9.ecloud-kdemo.com/rest/v1.0/pipelines?pipelineName=jhendrick_pipeline&projectName=Training_jhendrick" --insecure
+       name: cloudbees-flow-release
+```
+
+Compare the original build pack for maven found here: https://github.com/jenkins-x-buildpacks/jenkins-x-kubernetes/tree/master/packs/maven vs. our forked build pack.  We’ve removed all the references to skaffold, watch, and helm since we’re no longer having Jenkin’s X handle the deployment to our Kubernetes cluster.  We’ve also updated the pipeline file to make a call into our CloudBees Flow server:
+
+```bash
+curl -X POST --header "Authorization: Basic $(jx step credential -s flow-token -k token)" --header "Content-Type: application/json" --header "Accept: application/json" -d "{}" "https://ps9.ecloud-kdemo.com/rest/v1.0/pipelines?pipelineName=jhendrick_pipeline&projectName=Training_jhendrick" --insecure
+```
+
+The above API call into CloudBees Flow tells Flow to run a pipeline called jhendrick_pipeline within a project called Training_jhendrick.
+
+You’ll also notice that we’re using a Jenkins X feature (jx step credential) to get our secret, flow-token, which we created previously so that we can authenticate to the Flow Rest API. Note that there are many other possible ways to call into the CloudBees Flow API’s besides cURL such as the command line tool ectool as well as python or groovy libraries.  Also note that for a production environment we would want to setup the proper certificates rather than using the --insecure parameter.
+
+Next, we need to tell Jenkins X to use our new build pack: 
+
+```bash
+jx edit buildpack -u https://github.com/jhendrickCB/jenkins-x-kubernetes -r master -b
+```
+
+{{< image src="/img/jenkins-x-flow-integration/jx-edit-buildpack.jpg" alt="Jenkins X Editing the Current Build Pack" >}}
+
+Finally, we’re going to use [cURL](https://curl.haxx.se/) to call CloudBees Flow.  Since we have to authenticate when calling the [Flow REST API](http://docs.electric-cloud.com/eflow_doc/9_0/API/HTML/FlowAPI_Guide_9_0.htm), we’ll create a Kubernetes secret to store our username/password authentication token:
+
+```bash
+apiVersion: v1
+kind: Secret
+metadata:
+ name: flow-token
+type: Opaque
+data:
+ token: <Basic Auth Token>
+```
+
+Note: In this case, the <Basic Auth Token> will take the form of Username:Password base64 encoded.  Take note that we’ll actually need to base64 encode our Username:Password token twice as it will get base64 decoded automatically when we access it later.
```

Comment:
<Basic Auth Token>  doesn't appear in the actual post - probably a markdown issue.